src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -243,12 +243,7 @@ protected virtual void Initialize(ScriptSettingsManager settingsManager)
             string debugSentinelFileName = Path.Combine(hostLogPath, ScriptConstants.DebugSentinelFileName);
             this.LastDebugNotify = File.GetLastWriteTime(debugSentinelFileName);
 
-            IMetricsLogger metricsLogger = ScriptConfig.HostConfig.GetService<IMetricsLogger>();
-            if (metricsLogger == null)
-            {
-                metricsLogger = new MetricsLogger();
-                ScriptConfig.HostConfig.AddService<IMetricsLogger>(metricsLogger);
-            }
+            IMetricsLogger metricsLogger = CreateMetricsLogger();
```

Comment:
FYI -- I chose to move this out to a separate method b/c the `Initialize` method had hit its cyclomatic complexity limit.